photon-api/src/test/scala/com/linkedin/photon/ml/data/LocalDataSetTest.scala

```diff
@@ -132,4 +382,77 @@ class LocalDataSetTest {
       filteredDataPointsAll
         .forall(dataPoint => dataPoint._2.features.activeKeysIterator.toSet == Set(0, 1, 2, 3, 4, 5)))
   }
+
+  /**
+   * Test the Ratio Lower Bound computation
+   */
+  @Test(groups = Array[String]("testComputeRatioCILowerBound", "testCore"),
+    dataProvider = "dataForRandomEffectFeatureSelection")
+  def testComputeRatioCILowerBound(
+    features: Array[Vector[Double]],
+    labels: Array[Double],
+    expected: Map[Int, Double],
+    globalStats: BasicStatisticalSummary,
+    binaryIndices: Set[Int],
+    nonBinaryIndices: Set[Int],
+    globalPositiveInstances: Array[Double],
+    selectedFeatureNum: Int): Unit = {
+    // 8 columns of features
+    // col 1 for x = 0, y = 0
+    // col 2 for x = 0, y != 0
+    // col 3 for nonBinary
+    // col 4 for x = m, y = n
+    // col 5 for t > 1 not selected
+    // col 6 for t > 1 selected
+    // col 7 for t < 1 not selected
+    // col 8 for t < 1 selected
+
+    val labelAndFeatures = labels.zip(features)
+    val computed = LocalDataSet.computeRatioCILowerBound(
+      labelAndFeatures,
+      2.575,
+      globalStats,
+      globalPositiveInstances,
+      binaryIndices)
+
+    computed.foreach { case (key, value) =>
+      assertEquals(
+        expected(key).asInstanceOf[Double],
+        value,
```

Comment:
Should swap the order of arguments - the first argument should be the actual value, the second argument should be the expected value.

Also: unnecessary cast, `expected(key)` is already a `Double`